packages/core/content-manager/server/src/controllers/relations.ts

```diff
@@ -45,117 +43,196 @@ const sanitizeMainField = (model: any, mainField: any, userAbility: any) => {
+      // to the ctx
+      return;
+    }
+
+    const {
+      targetField,
+      fieldsToSelect,
+      mainField,
+      sourceSchema: { uid: sourceUid },
+      targetSchema: { uid: targetUid },
+      currentEntityId,
+    } = validation;
+
+    const { idsToOmit, idsToInclude, _q, ...query } = ctx.request.query;
 
     const permissionChecker = getService('permission-checker').create({
-      userAbility,
-      model: targetedModel.uid,
+      userAbility: ctx.state.userAbility,
+      model: targetUid,
     });
     const permissionQuery = await permissionChecker.sanitizedQuery.read(query);
 
     const queryParams = {
       sort: mainField,
-      fields: fieldsToSelect, // cannot select other fields as the user may not have the permissions
+      // cannot select other fields as the user may not have the permissions
+      fields: fieldsToSelect,
       ...permissionQuery,
     };
 
-    if (!isEmpty(idsToOmit)) {
-      addFiltersClause(queryParams, { id: { $notIn: idsToOmit } });
+    const isPublished = status === 'published';
+    // Only ever findAvailable (e.g. populate relation select) on a draft entry
+    addFiltersClause(queryParams, {
+      [PUBLISHED_AT_ATTRIBUTE]: isPublished ? { $ne: null } : null,
+    });
+
+    // We are looking for available content type relations and should be
+    // filtering by valid documentIds only
+    const stringIdsToOmit = idsToOmit?.filter((id: any) => !Number(id));
+    if (stringIdsToOmit?.length > 0) {
+      addFiltersClause(queryParams, {
+        documentId: { $notIn: uniq(idsToOmit) },
+      });
     }
 
-    // searching should be allowed only on mainField for permission reasons
     if (_q) {
+      // searching should be allowed only on mainField for permission reasons
       const _filter = isOperatorOfType('where', query._filter) ? query._filter : '$containsi';
       addFiltersClause(queryParams, { [mainField]: { [_filter]: _q } });
     }
 
-    if (entityId) {
-      const subQuery = strapi.db.queryBuilder(modelSchema.uid);
+    if (currentEntityId) {
+      // If we have been given an entity id, we need to filter out the
+      // relations that are already linked to the current entity
+      const subQuery = strapi.db.queryBuilder(sourceUid);
 
+      // The alias refers to the DB table of the target model
       const alias = subQuery.getAlias();
 
-      const where = {
-        id: entityId,
+      const where: Record<string, any> = {
+        id: currentEntityId,
         [`${alias}.id`]: { $notNull: true },
+        // Always find available draft entries, as we will be potentially
+        // connecting them from the CM edit view
+        [`${alias}.published_at`]: isPublished ? { $notNull: true } : { $notNull: false },
       };
 
-      if (!isEmpty(idsToInclude)) {
-        where[`${alias}.id`].$notIn = idsToInclude;
+      const stringIdsToInclude = idsToInclude?.filter((id: any) => !Number(id));
+      if ((stringIdsToInclude?.length ?? 0) !== 0) {
```

Comment:
is it necessary to do the parsing to be a number? If we say ids to include will always be a document Id I feel that might not be necessary

Reply:
You are right, I will remove all the Number filtering and update the contracts